### PR TITLE
fix: update docs for ApplicationService.Get to reflect reality

### PR DESCRIPTION
**Pull request by djeebus**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: server/application/application.proto [hunk 1]
```diff
@@ -21,7 +21,7 @@ import "github.com/argoproj/argo-cd/v2/reposerver/repository/repository.proto";
 message ApplicationQuery {
 	// the application's name
 	optional string name = 1;
-	// forces application reconciliation if set to true
+	// forces application reconciliation if set to 'hard'
```
Comment (crenshaw-dev) [inline server/application/application.proto:24]:
```suggestion
	// forces application reconciliation if set to 'normal' or 'hard' - 'hard' ignores any repo cache
```

Maybe a bit more precise?

https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go#L289-L292

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: server/application/application.proto (hunk 1 above)
Comment (djeebus) [reply server/application/application.proto:24]: If my reading of [the code](https://github.com/argoproj/argo-cd/blob/master/server/application/application.go#L665-L668) is correct, it defaults to "normal", and only switches if you send "hard". Do we want the text to reflect the functionality, or hide the implementation and describe the API instead?

Comment (crenshaw-dev) [reply server/application/application.proto:24]: Good read, you're right. Let's stick with your wording.

**Review by crenshaw-dev — APPROVED** [review]
Thanks, @djeebus!

**Pull request merged**